### PR TITLE
feat(Splitter): add Splitter component

### LINKED ISSUE #4715: Splitter component
New component! https://github.com/patternfly/patternfly/pull/3407

@jessiehuff to add keyboard interaction details

### PR DESCRIPTION
**Pull request by nicolethoen**
**What**: Closes #4715 

New issues created:
- Create examples which incorporate sliding interaction using mouse or keyboard to slide the splitter. - #5049
-- _We may decide we need to address #5049 before we can close this PR_
- Add Drawer and MasterDetails demos(?)/variants(?) which incorporate a sliding splitter. - #5050

### CODE REVIEW

**Review by mcarrano — DISMISSED** [review]
This looks good to me and thanks for opening the follow-on issues. I'm OK to merge this with the caveat that we have more work to do on adding examples if others agree.

**Review by mcoker — APPROVED** [review]
LGTM!

**Review by kmcfaul — DISMISSED** [review]

File: packages/react-core/src/components/Splitter/Splitter.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import styles from '@patternfly/react-styles/css/components/Splitter/splitter';
+import { css } from '@patternfly/react-styles';
+
+export enum SplitterOrientation {
+  horizontal = 'horizontal',
+  vertical = 'vertical'
+}
+
+export interface SplitterProps extends React.HTMLProps<HTMLDivElement> {
+  /** Additional classes added to the Splitter. */
```
Comment (tlabaj) [inline packages/react-core/src/components/Splitter/Splitter.tsx:11]: Nit. Sentence case

**Review by tlabaj — CHANGES_REQUESTED** [review]

**Review by tlabaj — APPROVED** [review]
LGTM

**Review by dlabrecq — APPROVED** [review]

**Pull request merged**